pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -545,19 +551,19 @@ func (c *Controller) deleteNode(logger klog.Logger, obj interface{}) {
 	c.checkNodeTopologyDistribution(logger)
 }
 
-// checkNodeTopologyDistribution updates Nodes in the topology cache and then
-// queues any Services that are past the threshold.
+// checkNodeTopologyDistribution updates Nodes in the proportionalCPUTopology
+// cache and then queues any Services that are past the threshold.
 func (c *Controller) checkNodeTopologyDistribution(logger klog.Logger) {
-	if c.topologyCache == nil {
+	if c.proportionalCPUTopology == nil {
```

Comment:
Non-blocking: It seems like it would be ideal to abstract this away into some kind of interface that any other theoretical proportional heuristics could hook into. With that said, the current approach still requires running a custom EndpointSlice controller and I don't think the proportional approach has been very useful, so I don't think it's worth adding too much complexity here for some theoretical future value. (You've likely already had a similar thought process, just writing it out for the record).

Reply:
agree on this thought